pkg/controller/workflow/status.go

```diff
@@ -0,0 +1,43 @@
+package workflow
+
+import (
+	"reflect"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Status is a mutable container containing the status of some particular reconciliation. It is expected to be updated
+// by a controller and at any time it reflects the status of the reconciled resource. Its state should fully match the
+// state of the resource so the information could be used to update the status field of the Custom Resource.
+type Status struct {
+	options    []status.Option
+	conditions []status.Condition
+}
+
+func (s *Status) EnsureCondition(condition status.Condition) {
+	condition.LastTransitionTime = metav1.Now()
+	for i, c := range s.conditions {
+		if c.Type == condition.Type {
+			// We don't update the last transition time in case status hasn't changed.
+			if s.conditions[i].Status == condition.Status {
+				condition.LastTransitionTime = s.conditions[i].LastTransitionTime
+			}
+			s.conditions[i] = condition
+			return
+		}
+	}
+	// Condition not found - appending
+	s.conditions = append(s.conditions, condition)
+}
+
+func (s *Status) EnsureOption(option status.Option) {
+	for i, c := range s.options {
+		if reflect.TypeOf(c) == reflect.TypeOf(option) {
```

Comment:
This will not work correctly with the new approach - it will be true for all `AtlasProjectStatusOption`s.

I suggest we scrap this check and just append every time - the last option will take precedence anyway.

Reply:
good catch.
Yes, so far it looks quite safe as the last option will override what was done by all previous ones (unless there's some conditional logic inside the function so the changes are done to some different parts).

If we get to this problem we may turn the model to maps instead or make Option a struct with name and func fields (so we'll be able to map by name)